website/docs/r/cloudfront_invalidation.html.markdown

```diff
@@ -0,0 +1,47 @@
+---
+subcategory: "CloudFront"
+layout: "aws"
+page_title: "AWS: aws_cloudfront_invalidation"
+description: |-
+  Terraform resource for managing an AWS CloudFront Invalidation.
+---
+
+# Resource: aws_cloudfront_invalidation
+
+Terraform resource for managing an AWS CloudFront Invalidations.
+
+## Example Usage
+
+### Basic Usage
+
+```terraform
+resource "aws_cloudfront_invalidation" "example" {
+  distribution_id = aws_cloudfront_distribution.example.id
+  paths = [
+    "/*",
+  ]
+}
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+* `distribution_id` - (Required) CloudFront Distribution ID.
+* `paths` - (Required) List of paths to invalidate.
+* `triggers` (Optional) Map of arbitrary keys and values that, when changed, will trigger a re-invocation. To force a re-invocation without changing these keys/values, use the [`terraform taint` command](https://www.terraform.io/docs/commands/taint.html).
```

Comment:
Could we provide an example of using triggers? And maybe more information about what happens if no triggers are specified. I assume if no triggers are specified the invalidation will be performed 1 time ever (on initial "resource create"), but triggers would make it happen basically on demand (that is awesome, btw).

Reply:
The idea and implementation is a copy of this https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_invocation#dynamic-invocation-example-using-triggers

Their docs give great examples.

Reply:
I mean more like, provide an example on the http example page in this PR. Thank you for the link! I want to learn more about this, I'm a n00b with aspirations of knowing what you're doing in this PR one day. :)